env vignette explains that no extra work needed when using from outer function (#6939)
Rdatatable/data.table: vignettes/datatable-programming.Rmd

In case of any doubts on the `env` interface functioning, set `verbose = TRUE` to inspect how expressions are resolved internally.

### Use `env` argument from inside another function

> It was a design decision that `env` argument will follow _Standard Evaluation_ rules. As a result, it does not require any special handling when being used from inside another function. As well it does not recognize data.table's `.()` alias for a `list()`.

Reviewer: is "standard evaluation" a well-defined term? An external reference (e.g. an R manual where this is discussed) would be ideal.

Author: precisely NSE is not in R docs, but I think linking this is the best resource on the topic: https://cran.r-project.org/doc/manuals/r-release/R-lang.html#Computing-on-the-language

> It was a design decision that `env` argument will follow _Standard Evaluation_ rules. As a result, it does not require any special handling when being used from inside another function. As well it does not recognize data.table's `.()` alias for a `list()`.

Reviewer: ```suggestion
It was a design decision that `env` argument will follow _Standard Evaluation_ rules. As a result, it does not require any special handling when being used from inside another function. Importantly, it does not recognize data.table's `.()` alias for a `list()`. You can add `. = list` to your `env` if you'd like.
```

Author: > You can add `. = list` to your `env` if you'd like.

this is wrong, it would only subtitute .() to list() inside i/j/by, and we talk here about env arg. Coming edit will make it clear.

```{r env_se}

> fun = function(x, col.mean) {
>   stopifnot(is.character(col.mean), is.data.table(x))
>   x[, mean(.col), env = list(.col = col.mean)]
> }
> fun(DT, col.mean="Petal.Length")

Reviewer: ```suggestion
fun = function(x, col.mean) {
  stopifnot(is.character(col.mean), is.data.table(x))
  x[, .(col_avg = mean(.col)), env = list(.col = col.mean, . = list)]
}
fun(DT, col.mean="Petal.Length")
```

Author: using `. = list` is useless, in i/j/by `.()` will be properly substituted. New commit explains that `env = .(.col="v1")` will not work. `.()` in i/j/by is not affected by `env`

Reviewer: Got it, that's good. Thanks for clarifying.

```

If the outer function itself follows NSE (Non-Standard Evaluation) rules, then it has to resolve language objects the same way as when passing it's arguments to any other function.

> ```{r nse}

Reviewer: ```suggestion
```{r env_nse}
```

>   x[, mean(.col), env = list(.col = col.mean)]

Reviewer: ```suggestion
  x[, .(col_avg = mean(.col)), env = list(.col = col.mean)]
```

Reviewer: I think using `.` here can help clarify that `.` is still OK in other arguments
